ygen/protogen.go

```diff
@@ -500,6 +502,24 @@ func writeProtoEnums(enums map[string]*yangEnum) ([]string, []error) {
 				continue
 			}
 			p.Values = ge.Values
+
+			// If the supplied enum entry has the valuePrefix annotation then use it to
+			// calculate the enum value names.
+			if e, ok := enum.entry.Annotation["valuePrefix"]; ok {
+				t, ok := e.([]string)
+				if ok {
+					pp := []string{}
+					for _, pe := range t {
+						pp = append(pp, safeProtoIdentifierName(yang.CamelCase(pe)))
+					}
+					p.ValuePrefix = strings.Join(pp, "_")
+				}
+			}
+
+			if p.ValuePrefix == "" {
+				p.ValuePrefix = enum.name
```

Comment:
Seems like you could just set this by default above the check for Annotation.

Reply:
Done.